### PR TITLE
IR-96: Create model to store highlights/notes/goals

### PR DESCRIPTION
**Pull request by ishugupta92**

### CODE REVIEW

File: apps/retrospective/models/sprint_extra.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// SprintExtraType
+const (
+	NoteType SprintExtraType = iota
+	HighlightType
+	GoalType
+)
+
+// SprintExtra represent Goals, Highlights and Notes of a sprint
+type SprintExtra struct {
+	gorm.Model
```
Comment (anmolghosh) [inline apps/retrospective/models/sprint_extra.go:54]: validate usage of default `createdAt`

File: apps/retrospective/models/sprint_member_task.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ const (
 type SprintMemberTask struct {
 	gorm.Model
 	SprintMember     SprintMember
-	SprintMemberID   uint                 `gorm:"not null"`
+	SprintMemberID   uint `gorm:"not null"`
```
Comment (anmolghosh) [inline apps/retrospective/models/sprint_member_task.go:42]: formatting issue

Comment (anmolghosh) [reply apps/retrospective/models/sprint_member_task.go:42]: trailing space in
```
// MemberTaskRole
const (
	Implementor MemberTaskRole = iota
	Reviewer    
	Validator   
)
```

File: apps/retrospective/models/sprint_extra.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	Retrospective   Retrospective
+	RetrospectiveID uint             `gorm:"not null"`
+	Text            string           `gorm:"type:text; not null"`
+	Scope           SprintExtraScope `gorm:"default:0; not null"`
+	AssigneeID      uint
+	Assignee        userModels.User
+	ResolvedAt      *time.Time
+	ExpectedAt      *time.Time
+	CreatedByID     uint `gorm:"not null"`
+	CreatedBy       userModels.User
```
Comment (anmolghosh) [inline apps/retrospective/models/sprint_extra.go]: register this model in admin. Override for choice fields

**Review by anmolghosh — CHANGES_REQUESTED** [review]

File: apps/retrospective/models/sprint_extra.go (hunk 1 above)
Comment (ishugupta92) [reply apps/retrospective/models/sprint_extra.go:54]: Added a new field AddedAt as discussed

File: apps/retrospective/models/sprint_member_task.go (hunk 2 above)
Comment (ishugupta92) [reply apps/retrospective/models/sprint_member_task.go:42]: Formatting is as suggested by GoLint

Also, I cannot find any trailing space in this file.

File: apps/retrospective/models/sprint_extra.go (hunk 3 above)
Comment (ishugupta92) [reply apps/retrospective/models/sprint_extra.go]: Will do this in a separate PR as discussed.

**Review by anmolghosh — APPROVED** [review]

**Pull request merged**